#!/usr/bin/env python
from dynamic_reconfigure.parameter_generator_catkin import *
PACKAGE = "boxing"

gen = ParameterGenerator()

gen.add(
    "voxel_size",                       # name
    double_t,                           # type
    0,                                  # config level
    "Specifies the voxel size.",        # description
    0.2,                                # default
    0.0001,                             # min
    10                                  # max
)

gen.add(
    "m_filter_by_min_score_enabled",
    bool_t,
    0,
    "Enable filtering by minimal score.",
    True
)

gen.add(
    "filter_by_min_score",
    double_t,
    0,
    "Minimal score for filtering.",
    0.005,
    0.0001,
    10
)

gen.add(
    "sor_enabled",
    bool_t,
    0,
    "Enable SOR.",
    False
)

gen.add(
    "sor_mean_k",
    double_t,
    0,
    "The mean K for SOR.",
    2,
    0.0001,
    10
)

gen.add(
    "sor_stddev_mul_thresh",
    double_t,
    0,
    "The standard derivation of multiple threshholds for SOR.",
    3,
    0.0001,
    10
)

exit(gen.generate(PACKAGE, "boxing", "boxing"))
